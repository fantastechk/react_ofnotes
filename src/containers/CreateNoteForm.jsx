import React, { useEffect } from 'react';
import { useHistory } from 'react-router-dom';
import { makeStyles } from '@material-ui/styles';
import { useFormik } from 'formik';
import SaveIcon from '@material-ui/icons/Save';

import { useDispatch, useSelector } from 'react-redux';
import { selectPanelError } from '../store/selectors';
import * as actions from '../store/actions';

import NoteForm from '../components/NoteForm';
import { NoteFormButton } from '../components/NoteFormInputs';


const useStyles = makeStyles(theme => ({
  icon: {
    marginLeft: theme.spacing()
  }
}))

const CreateNoteForm = props => {
  const classes = useStyles();
  const dispatch = useDispatch();
  const history = useHistory();
  const panelError = useSelector(selectPanelError)

  const formik = useFormik({
    initialValues: {
      title: '',
      md: '',
      tags: []
    },
    onSubmit: values => {
      dispatch(actions.createNote(values, slug =>{
        history.push(`/${slug}`)
      }))
    },
    validate: values => {
      const errors = {};
      if (!values.title) {
        errors.title = 'Title is required!'
      }
      if(!values.md) {
        errors.md = 'Note is required!'
      }
      if (!Array.isArray(values.tags)) {
        errors.tags = 'Something went wrong here..'
      }
      return errors;
    }
  })

  // if we are on this page, there should not be a current
  // not selected in the store. do this whenever this
  // component mounts
  useEffect(() => {
    dispatch(actions.setCurrentNote(null))
  }, [dispatch])

  return (
    <NoteForm {...formik}
              panelError={panelError}
              formActions={
                <NoteFormButton
                  type="submit"
                  aria-label="save note"
                  >
                   Save <SaveIcon className={classes.icon} />
                </NoteFormButton>
              }
              {...props}
              />
  )
}

export default CreateNoteForm;
