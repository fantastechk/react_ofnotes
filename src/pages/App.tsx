import React from "react";
import { Route, Switch } from "react-router-dom";
import { Layout } from "components";
import { Home, Note, Form } from "pages";
import {
  ProvideThemeContext,
  ProvidSideBarContext,
  ProvideNoteContext,
} from "hooks";

import moment from "moment";
import { HashRouter as Router } from "react-router-dom";

moment.updateLocale("en", {
  relativeTime: {
    s: "1s",
    ss: "%ds",
    m: "1m",
    mm: "%dm",
    h: "1h",
    hh: "%dh",
    d: "1d",
    dd: "%dd",
    y: "1y",
    yy: "%dy",
  },
});

export const App = () => (
  <Router>
    <ProvideNoteContext>
      <ProvideThemeContext>
        <ProvidSideBarContext>
          <Layout>
            <Switch>
              <Route exact path="/" component={Home} />
              <Route exact path="/create" component={Form} />
              <Route exact path="/:slug/edit" component={Form} />
              <Route exact path="/:slug" component={Note} />
            </Switch>
          </Layout>
        </ProvidSideBarContext>
      </ProvideThemeContext>
    </ProvideNoteContext>
  </Router>
);
