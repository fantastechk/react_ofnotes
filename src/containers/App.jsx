import React from 'react';
import { Route,
         Switch } from 'react-router-dom';
import { makeStyles } from '@material-ui/styles';

import { useDispatch, useSelector } from 'react-redux';
import { selectMenuOpen } from '../store/selectors';
import * as actions from '../store/actions';

import SideBar from './SideBar';
import CreateNoteForm from './CreateNoteForm';
import EditNoteForm from './EditNoteForm';
import NoteDisplay from './NoteDisplay';


const useStyles = makeStyles({
  root: {
    display: 'flex',
    width: '100%',
    height: '100%'
  }
});


const App = () => {
  const classes = useStyles();
  const dispatch = useDispatch();
  const menuExpanded = useSelector(selectMenuOpen)

  const toggleMenuOpen = e => {
    dispatch(actions.toggleMenu())
  }

  return (
    <div className={classes.root}>
      <SideBar expanded={menuExpanded}
               toggleMenu={toggleMenuOpen} />

      <Switch>
        <Route exact path="/">
          <CreateNoteForm toggleMenu={toggleMenuOpen} />
        </Route>
        <Route path="/:slug/edit">
          <EditNoteForm toggleMenu={toggleMenuOpen} />
        </Route>
        <Route path="/:slug">
          <NoteDisplay toggleMenu={toggleMenuOpen} />
        </Route>
      </Switch>
    </div>
  )
}

export default App;
