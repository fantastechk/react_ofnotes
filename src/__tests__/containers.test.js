import React from 'react';

import { render } from '../test-utils';
import { fireEvent, wait } from '@testing-library/react';

import sinon from 'sinon';
import queries from '../db/queries';
import * as actions from '../store/actions';
import * as selectors from '../store/selectors';
import * as reactRedux from 'react-redux';

import App from '../containers/App';
import CreateNoteForm from '../containers/CreateNoteForm';
import EditNoteForm from '../containers/EditNoteForm';
import NoteDisplay from '../containers/NoteDisplay';
import SideBar from '../containers/SideBar';


describe('containers', () => {
  let dispatchSpy;

  beforeAll(() => {
    sinon.stub(queries)
    sinon.stub(reactRedux, 'useDispatch')
  })

  beforeEach(() => {
    sinon.reset()
    dispatchSpy = sinon.spy()
    reactRedux.useDispatch.returns(dispatchSpy)
  })

  describe('App', () => {
    it('matches snapshot', () => {
      const { asFragment } = render(<App />)
      expect(asFragment()).toMatchSnapshot()
    })
  })

  describe('CreateNoteForm', () => {
    beforeAll(() => {
      sinon.stub(actions, 'setCurrentNote')
      sinon.stub(actions, 'createNote')
    })

    beforeEach(() => {
      actions.setCurrentNote.reset()
      actions.setCurrentNote.returns({
        type: 'TEST_SET_NOTE'
      })
      actions.createNote.reset()
      actions.createNote.returns({
        type: 'TEST_CREATE_NOTE'
      })
    })

    afterAll(() => {
      actions.setCurrentNote.restore()
      actions.createNote.restore()
    })

    it('matches snapshot', () => {
      const { asFragment } = render(<CreateNoteForm />)
      expect(asFragment()).toMatchSnapshot()
    })

    it('initializes form with empty data', () => {
      const { getByLabelText } = render(<CreateNoteForm />)
      expect(getByLabelText('Title')).toHaveValue('')
      expect(getByLabelText('Tags')).toHaveValue('')
      expect(getByLabelText('Note')).toHaveValue('')
    })

    it('sets current note to null on render', () => {
      render(<CreateNoteForm />)
      sinon.assert.calledOnce(reactRedux.useDispatch)
      sinon.assert.calledOnce(actions.setCurrentNote)
      sinon.assert.calledOnce(dispatchSpy)
      sinon.assert.calledWithExactly(dispatchSpy, {
        type: 'TEST_SET_NOTE'
      })
    })

    it('updates form state on change', async () => {
      const { getByLabelText } = render(<CreateNoteForm />)
      fireEvent.change(getByLabelText('Title'), { 
        target: { value: 'test val' }
      })
      await wait(() => {
        expect(getByLabelText('Title')).toHaveValue('test val')
      })
    })

    it('validates the title', async () => {
      const { getByLabelText, getByText } = render(<CreateNoteForm />)
      fireEvent.blur(getByLabelText('Title'))
      await wait(() => {
        expect(getByText('Title is required!')).toBeInTheDocument()
      })
    })

    it('validates markdown', async () => {
      const { getByLabelText, getByText } = render(<CreateNoteForm />)
      fireEvent.blur(getByLabelText('Note'))
      await wait(() => {
        expect(getByText('Note is required!')).toBeInTheDocument()
      })
    })

    it('dispatches create note on save', async () => {
      const { getByTestId, getByLabelText } = render(<CreateNoteForm />)
      fireEvent.change(getByLabelText('Title'), {
        target: { value: 'test' }
      })
      fireEvent.change(getByLabelText('Note'), {
        target: { value: 'test note' }
      })
      fireEvent.submit(getByTestId("form"))
      await wait(() => {
        sinon.assert.calledOnce(actions.createNote)
        sinon.assert.calledWith(actions.createNote, {
          title: 'test',
          md: 'test note',
          tags: []
        })
        sinon.assert.calledTwice(dispatchSpy)
      })
    })

  })

  describe('EditNoteForm', () => {
    beforeAll(() => {
      sinon.stub(actions, 'setCurrentNote')
      sinon.stub(actions, 'editNote')
      sinon.stub(actions, 'deleteNote')
      sinon.stub(selectors, 'selectCurrentNote')
    })
    
    beforeEach(() => {
      actions.setCurrentNote.reset()
      actions.setCurrentNote.returns({
        type: 'TEST_SET_NOTE'
      })
      actions.editNote.reset()
      actions.editNote.returns({
        type: 'TEST_CREATE_NOTE'
      })
      actions.deleteNote.reset()
      actions.deleteNote.returns({
        type: 'TEST_DELETE_NOTE'
      })
      selectors.selectCurrentNote.reset()
      selectors.selectCurrentNote.returns({
        title: 'test title',
        md: 'test markdown',
        tags: ['testtag', 'testtag2'],
        id: 42
      })
    })

    afterAll(() => {
      actions.setCurrentNote.restore()
      actions.editNote.restore()
      selectors.selectCurrentNote.restore()
    })

    it('matches snapshot', () => {
      const { asFragment } = render(<EditNoteForm />)
      expect(asFragment()).toMatchSnapshot()
    })
  
    it('sets form values based on current note', async () => {
      const { getByLabelText } = render(<EditNoteForm />)
      await wait(() => {
        expect(getByLabelText('Title')).toHaveValue('test title')
        expect(getByLabelText('Note')).toHaveValue('test markdown')
        expect(getByLabelText('Tags')).toHaveValue('testtag testtag2')
      })
    })

    it('validates title required', async () => {
      const { getByLabelText, getByText } = render(<EditNoteForm />)
      fireEvent.blur(getByLabelText('Title'))
      fireEvent.change(getByLabelText('Title'), {
        target: { value: '' }
      })
      await wait(() => {
        expect(getByText('Title is required!')).toBeInTheDocument()
      })
    })

    it('validates note required', async () => {
      const { getByLabelText, getByText } = render(<EditNoteForm />)
      fireEvent.blur(getByLabelText('Note'))
      fireEvent.change(getByLabelText('Note'), {
        target: { value: '' }
      })
      await wait(() => {
        expect(getByText('Note is required!')).toBeInTheDocument()
      })
    })

    it('dispatches edit note on change', async () => {
      const { getByLabelText } = render(<EditNoteForm />)
      fireEvent.change(getByLabelText('Title'), {
        target: { value: 'test title changed' }
      })
      await wait(() => {
        sinon.assert.calledWithMatch(actions.editNote, 42, {
          title: 'test title changed'
        })
      })
    }, 6000)

    it('dispatches delete on delete click', async () => {
      const { getByText } = render(<EditNoteForm />)
      fireEvent.click(getByText('Delete', { exact: false }))
      await wait(() => {
        sinon.assert.calledOnce(actions.deleteNote)
        sinon.assert.calledWith(actions.deleteNote, 42)
        sinon.assert.calledWithExactly(dispatchSpy, {
          type: 'TEST_DELETE_NOTE'
        })
      })

    })
  })

  describe('NoteDisplay', () => {
    beforeAll(() => {
      sinon.stub(actions, 'setCurrentNote')
      sinon.stub(selectors, 'selectCurrentNote')
    })

    beforeEach(() => {
      actions.setCurrentNote.reset()
      actions.setCurrentNote.returns({
        type: 'TEST_SET_CURRENT_NOTE'
      })
      selectors.selectCurrentNote.reset()
      selectors.selectCurrentNote.returns({
        title: 'test title',
        tags: ['test1', 'test2'],
        md: 'asdfasdf',
        id: 42
      })
    })

    afterAll(() => {
      actions.setCurrentNote.restore()
      selectors.selectCurrentNote.restore()
    })

    it('matches snapshot', () => {
      const { asFragment } = render(<NoteDisplay />)
      expect(asFragment()).toMatchSnapshot()
    })

    it('calls set current note on mount', async () => {
      render(<NoteDisplay />)
      await wait(() => {
        sinon.assert.calledOnce(actions.setCurrentNote)
        // can't easily stub useParams to check that the
        // correct slig is dispatched
        sinon.assert.calledWithMatch(dispatchSpy, {
          type: 'TEST_SET_CURRENT_NOTE'
        })
      })
    })

  })

  describe('SideBar', () => {
    it('matches snapshot', () => {
      const { asFragment } = render(<SideBar />)
      expect(asFragment()).toMatchSnapshot()
    })
  })
})