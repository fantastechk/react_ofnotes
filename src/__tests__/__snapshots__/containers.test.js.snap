// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`containers App matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="makeStyles-root-1"
  >
    <div
      class="MuiPaper-root-7 makeStyles-root-2 makeStyles-root-6 MuiPaper-elevation1-10 MuiPaper-rounded-8"
    >
      <div
        class="makeStyles-root-35"
      >
        <a
          class="MuiAvatar-root-38 makeStyles-avatar-34 MuiAvatar-colorDefault-39"
          href="#/"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root-41"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M14 2H6c-1.1 0-1.99.9-1.99 2L4 20c0 1.1.89 2 1.99 2H18c1.1 0 2-.9 2-2V8l-6-6zm2 14h-3v3h-2v-3H8v-2h3v-3h2v3h3v2zm-3-7V3.5L18.5 9H13z"
            />
          </svg>
        </a>
        <div
          class="makeStyles-titleWrap-36"
        >
          <h5
            class="MuiTypography-root-50 makeStyles-brandFont-37 MuiTypography-h5-59"
          >
            Of Note
            <small>
              <em>
                s
              </em>
            </small>
          </h5>
          <h6
            class="MuiTypography-root-50 makeStyles-brandFont-37 MuiTypography-subtitle2-62 MuiTypography-colorTextSecondary-76"
          >
            Offline note taking
          </h6>
        </div>
        <button
          class="MuiButtonBase-root-89 MuiIconButton-root-80"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiIconButton-label-88"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root-41"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M20 15.31L23.31 12 20 8.69V4h-4.69L12 .69 8.69 4H4v4.69L.69 12 4 15.31V20h4.69L12 23.31 15.31 20H20v-4.69zM12 18c-3.31 0-6-2.69-6-6s2.69-6 6-6 6 2.69 6 6-2.69 6-6 6z"
              />
            </svg>
          </span>
          <span
            class="MuiTouchRipple-root-182"
          />
        </button>
      </div>
      <div
        class="makeStyles-filter-4"
      >
        <div
          class="MuiFormControl-root-95 MuiTextField-root-94 makeStyles-inp-92 MuiFormControl-fullWidth-98"
        >
          <label
            class="MuiFormLabel-root-111 MuiInputLabel-root-99 MuiInputLabel-formControl-105 MuiInputLabel-animated-108 MuiInputLabel-shrink-107 MuiInputLabel-outlined-110"
            data-shrink="true"
          >
            Filter
          </label>
          <div
            class="MuiInputBase-root-133 MuiOutlinedInput-root-118 MuiInputBase-fullWidth-142 MuiInputBase-formControl-134"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input-143 MuiOutlinedInput-input-127"
              name="filter"
              type="text"
              value=""
            />
            <fieldset
              aria-hidden="true"
              class="PrivateNotchedOutline-root-153 MuiOutlinedInput-notchedOutline-126"
              style="padding-left: 8px;"
            >
              <legend
                class="PrivateNotchedOutline-legend-154"
                style="width: 0px;"
              >
                <span>
                  ​
                </span>
              </legend>
            </fieldset>
          </div>
          <p
            class="MuiFormHelperText-root-155 MuiFormHelperText-contained-159"
          >
            Search tags or titles
          </p>
        </div>
      </div>
      <hr
        class="MuiDivider-root-163"
      />
      <div
        class="makeStyles-eh-5"
      >
        <ul
          class="MuiList-root-170 MuiList-padding-171"
        />
      </div>
    </div>
    <div
      class="makeStyles-root-177"
    >
      <form>
        <div
          class="makeStyles-root-178"
        >
          <button
            class="MuiButtonBase-root-89 MuiIconButton-root-80 makeStyles-mobileMenu-179"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiIconButton-label-88"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root-41"
                focusable="false"
                role="presentation"
                viewBox="0 0 24 24"
              >
                <path
                  d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
                />
              </svg>
            </span>
            <span
              class="MuiTouchRipple-root-182"
            />
          </button>
          <span
            class="MuiTypography-root-50 makeStyles-title-181 MuiTypography-h5-59"
          >
            New Note
          </span>
        </div>
        <div
          class="MuiFormControl-root-95 MuiTextField-root-94 makeStyles-inp-92 MuiFormControl-fullWidth-98"
        >
          <label
            class="MuiFormLabel-root-111 MuiInputLabel-root-99 MuiInputLabel-formControl-105 MuiInputLabel-animated-108 MuiInputLabel-outlined-110"
            data-shrink="false"
          >
            Title
          </label>
          <div
            class="MuiInputBase-root-133 MuiOutlinedInput-root-118 MuiInputBase-fullWidth-142 MuiInputBase-formControl-134"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input-143 MuiOutlinedInput-input-127"
              name="title"
              type="text"
              value=""
            />
            <fieldset
              aria-hidden="true"
              class="PrivateNotchedOutline-root-153 MuiOutlinedInput-notchedOutline-126"
              style="padding-left: 8px;"
            >
              <legend
                class="PrivateNotchedOutline-legend-154"
                style="width: 0.01px;"
              >
                <span>
                  ​
                </span>
              </legend>
            </fieldset>
          </div>
          <p
            class="MuiFormHelperText-root-155 MuiFormHelperText-contained-159"
          >
             
          </p>
        </div>
        <div
          class="MuiFormControl-root-95 MuiTextField-root-94 makeStyles-inp-92 MuiFormControl-fullWidth-98"
        >
          <label
            class="MuiFormLabel-root-111 MuiInputLabel-root-99 MuiInputLabel-formControl-105 MuiInputLabel-animated-108 MuiInputLabel-outlined-110"
            data-shrink="false"
          >
            Note
          </label>
          <div
            class="MuiInputBase-root-133 MuiOutlinedInput-root-118 MuiInputBase-fullWidth-142 MuiInputBase-formControl-134 MuiInputBase-multiline-141 MuiOutlinedInput-multiline-125"
          >
            <textarea
              aria-invalid="false"
              class="MuiInputBase-input-143 MuiOutlinedInput-input-127 undefined makeStyles-textarea-175 MuiInputBase-inputMultiline-148 MuiOutlinedInput-inputMultiline-130"
              name="md"
              rows="2"
            />
            <fieldset
              aria-hidden="true"
              class="PrivateNotchedOutline-root-153 MuiOutlinedInput-notchedOutline-126"
              style="padding-left: 8px;"
            >
              <legend
                class="PrivateNotchedOutline-legend-154"
                style="width: 0.01px;"
              >
                <span>
                  ​
                </span>
              </legend>
            </fieldset>
          </div>
          <p
            class="MuiFormHelperText-root-155 MuiFormHelperText-contained-159"
          >
            Supports Github flavored markdown
          </p>
        </div>
        <div
          class="MuiFormControl-root-95 MuiTextField-root-94 makeStyles-inp-92 MuiFormControl-fullWidth-98"
        >
          <label
            class="MuiFormLabel-root-111 MuiInputLabel-root-99 MuiInputLabel-formControl-105 MuiInputLabel-animated-108 MuiInputLabel-outlined-110"
            data-shrink="false"
          >
            Tags
          </label>
          <div
            class="MuiInputBase-root-133 MuiOutlinedInput-root-118 MuiInputBase-fullWidth-142 MuiInputBase-formControl-134"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input-143 MuiOutlinedInput-input-127"
              name="tags"
              type="text"
              value=""
            />
            <fieldset
              aria-hidden="true"
              class="PrivateNotchedOutline-root-153 MuiOutlinedInput-notchedOutline-126"
              style="padding-left: 8px;"
            >
              <legend
                class="PrivateNotchedOutline-legend-154"
                style="width: 0.01px;"
              >
                <span>
                  ​
                </span>
              </legend>
            </fieldset>
          </div>
          <p
            class="MuiFormHelperText-root-155 MuiFormHelperText-contained-159"
          >
            Separated by spaces
          </p>
        </div>
        <div />
      </form>
    </div>
  </div>
</DocumentFragment>
`;

exports[`containers CreateNoteForm matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="makeStyles-root-195"
  >
    <form>
      <div
        class="makeStyles-root-196"
      >
        <button
          class="MuiButtonBase-root MuiIconButton-root makeStyles-mobileMenu-197"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiIconButton-label"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
              />
            </svg>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
        <span
          class="MuiTypography-root makeStyles-title-199 MuiTypography-h5"
        >
          New Note
        </span>
      </div>
      <div
        class="MuiFormControl-root MuiTextField-root makeStyles-inp-251 MuiFormControl-fullWidth"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
          data-shrink="false"
        >
          Title
        </label>
        <div
          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input MuiOutlinedInput-input"
            name="title"
            type="text"
            value=""
          />
          <fieldset
            aria-hidden="true"
            class="PrivateNotchedOutline-root-312 MuiOutlinedInput-notchedOutline"
            style="padding-left: 8px;"
          >
            <legend
              class="PrivateNotchedOutline-legend-313"
              style="width: 0.01px;"
            >
              <span>
                ​
              </span>
            </legend>
          </fieldset>
        </div>
        <p
          class="MuiFormHelperText-root MuiFormHelperText-contained"
        >
           
        </p>
      </div>
      <div
        class="MuiFormControl-root MuiTextField-root makeStyles-inp-251 MuiFormControl-fullWidth"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
          data-shrink="false"
        >
          Note
        </label>
        <div
          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-multiline MuiOutlinedInput-multiline"
        >
          <textarea
            aria-invalid="false"
            class="MuiInputBase-input MuiOutlinedInput-input undefined makeStyles-textarea-193 MuiInputBase-inputMultiline MuiOutlinedInput-inputMultiline"
            name="md"
            rows="2"
          />
          <fieldset
            aria-hidden="true"
            class="PrivateNotchedOutline-root-312 MuiOutlinedInput-notchedOutline"
            style="padding-left: 8px;"
          >
            <legend
              class="PrivateNotchedOutline-legend-313"
              style="width: 0.01px;"
            >
              <span>
                ​
              </span>
            </legend>
          </fieldset>
        </div>
        <p
          class="MuiFormHelperText-root MuiFormHelperText-contained"
        >
          Supports Github flavored markdown
        </p>
      </div>
      <div
        class="MuiFormControl-root MuiTextField-root makeStyles-inp-251 MuiFormControl-fullWidth"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
          data-shrink="false"
        >
          Tags
        </label>
        <div
          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input MuiOutlinedInput-input"
            name="tags"
            type="text"
            value=""
          />
          <fieldset
            aria-hidden="true"
            class="PrivateNotchedOutline-root-312 MuiOutlinedInput-notchedOutline"
            style="padding-left: 8px;"
          >
            <legend
              class="PrivateNotchedOutline-legend-313"
              style="width: 0.01px;"
            >
              <span>
                ​
              </span>
            </legend>
          </fieldset>
        </div>
        <p
          class="MuiFormHelperText-root MuiFormHelperText-contained"
        >
          Separated by spaces
        </p>
      </div>
      <div />
    </form>
  </div>
</DocumentFragment>
`;

exports[`containers EditNoteForm matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="makeStyles-root-336"
  >
    <form>
      <div
        class="makeStyles-root-337"
      >
        <button
          class="MuiButtonBase-root MuiIconButton-root makeStyles-mobileMenu-338"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiIconButton-label"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
              />
            </svg>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
        <span
          class="MuiTypography-root makeStyles-title-340 MuiTypography-h5"
        >
          New Note
        </span>
      </div>
      <div
        class="MuiFormControl-root MuiTextField-root makeStyles-inp-392 MuiFormControl-fullWidth"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
          data-shrink="false"
        >
          Title
        </label>
        <div
          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input MuiOutlinedInput-input"
            name="title"
            type="text"
            value=""
          />
          <fieldset
            aria-hidden="true"
            class="PrivateNotchedOutline-root-453 MuiOutlinedInput-notchedOutline"
            style="padding-left: 8px;"
          >
            <legend
              class="PrivateNotchedOutline-legend-454"
              style="width: 0.01px;"
            >
              <span>
                ​
              </span>
            </legend>
          </fieldset>
        </div>
        <p
          class="MuiFormHelperText-root MuiFormHelperText-contained"
        >
           
        </p>
      </div>
      <div
        class="MuiFormControl-root MuiTextField-root makeStyles-inp-392 MuiFormControl-fullWidth"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
          data-shrink="false"
        >
          Note
        </label>
        <div
          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-multiline MuiOutlinedInput-multiline"
        >
          <textarea
            aria-invalid="false"
            class="MuiInputBase-input MuiOutlinedInput-input undefined makeStyles-textarea-334 MuiInputBase-inputMultiline MuiOutlinedInput-inputMultiline"
            name="md"
            rows="2"
          />
          <fieldset
            aria-hidden="true"
            class="PrivateNotchedOutline-root-453 MuiOutlinedInput-notchedOutline"
            style="padding-left: 8px;"
          >
            <legend
              class="PrivateNotchedOutline-legend-454"
              style="width: 0.01px;"
            >
              <span>
                ​
              </span>
            </legend>
          </fieldset>
        </div>
        <p
          class="MuiFormHelperText-root MuiFormHelperText-contained"
        >
          Supports Github flavored markdown
        </p>
      </div>
      <div
        class="MuiFormControl-root MuiTextField-root makeStyles-inp-392 MuiFormControl-fullWidth"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
          data-shrink="false"
        >
          Tags
        </label>
        <div
          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input MuiOutlinedInput-input"
            name="tags"
            type="text"
            value=""
          />
          <fieldset
            aria-hidden="true"
            class="PrivateNotchedOutline-root-453 MuiOutlinedInput-notchedOutline"
            style="padding-left: 8px;"
          >
            <legend
              class="PrivateNotchedOutline-legend-454"
              style="width: 0.01px;"
            >
              <span>
                ​
              </span>
            </legend>
          </fieldset>
        </div>
        <p
          class="MuiFormHelperText-root MuiFormHelperText-contained"
        >
          Separated by spaces
        </p>
      </div>
      <div />
    </form>
  </div>
</DocumentFragment>
`;

exports[`containers NoteDisplay matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="makeStyles-root-478"
  >
    <div
      class="makeStyles-root-479"
    >
      <button
        class="MuiButtonBase-root MuiIconButton-root makeStyles-mobileMenu-480"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiIconButton-label"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
            />
          </svg>
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
      <span
        class="MuiTypography-root makeStyles-title-482 MuiTypography-h5"
      />
    </div>
    <div
      class="makeStyles-content-473"
    >
      <span />
    </div>
    <div
      class="makeStyles-tags-476"
    />
    <div />
  </div>
</DocumentFragment>
`;

exports[`containers SideBar matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiPaper-root makeStyles-root-544 makeStyles-root-548 MuiPaper-elevation1 MuiPaper-rounded"
  >
    <div
      class="makeStyles-root-577"
    >
      <a
        class="MuiAvatar-root makeStyles-avatar-576 MuiAvatar-colorDefault"
        href="#/"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root"
          focusable="false"
          role="presentation"
          viewBox="0 0 24 24"
        >
          <path
            d="M14 2H6c-1.1 0-1.99.9-1.99 2L4 20c0 1.1.89 2 1.99 2H18c1.1 0 2-.9 2-2V8l-6-6zm2 14h-3v3h-2v-3H8v-2h3v-3h2v3h3v2zm-3-7V3.5L18.5 9H13z"
          />
        </svg>
      </a>
      <div
        class="makeStyles-titleWrap-578"
      >
        <h5
          class="MuiTypography-root makeStyles-brandFont-579 MuiTypography-h5"
        >
          Of Note
          <small>
            <em>
              s
            </em>
          </small>
        </h5>
        <h6
          class="MuiTypography-root makeStyles-brandFont-579 MuiTypography-subtitle2 MuiTypography-colorTextSecondary"
        >
          Offline note taking
        </h6>
      </div>
      <button
        class="MuiButtonBase-root MuiIconButton-root"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiIconButton-label"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M20 15.31L23.31 12 20 8.69V4h-4.69L12 .69 8.69 4H4v4.69L.69 12 4 15.31V20h4.69L12 23.31 15.31 20H20v-4.69zM12 18c-3.31 0-6-2.69-6-6s2.69-6 6-6 6 2.69 6 6-2.69 6-6 6z"
            />
          </svg>
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </div>
    <div
      class="makeStyles-filter-546"
    >
      <div
        class="MuiFormControl-root MuiTextField-root makeStyles-inp-634 MuiFormControl-fullWidth"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined"
          data-shrink="true"
        >
          Filter
        </label>
        <div
          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input MuiOutlinedInput-input"
            name="filter"
            type="text"
            value=""
          />
          <fieldset
            aria-hidden="true"
            class="PrivateNotchedOutline-root-695 MuiOutlinedInput-notchedOutline"
            style="padding-left: 8px;"
          >
            <legend
              class="PrivateNotchedOutline-legend-696"
              style="width: 0px;"
            >
              <span>
                ​
              </span>
            </legend>
          </fieldset>
        </div>
        <p
          class="MuiFormHelperText-root MuiFormHelperText-contained"
        >
          Search tags or titles
        </p>
      </div>
    </div>
    <hr
      class="MuiDivider-root"
    />
    <div
      class="makeStyles-eh-547"
    >
      <ul
        class="MuiList-root MuiList-padding"
      />
    </div>
  </div>
</DocumentFragment>
`;
